src/manifold/src/quickhull.cpp

```diff
@@ -70,104 +66,127 @@ std::pair<manifold::Vec<manifold::Halfedge>,std::vector<glm::dvec3>> QuickHull::
 
   std::vector<size_t> halfEdgeStack;
   std::vector<bool> halfEdgeProcessed(mesh.halfEdges.size(), false);
-  std::vector<bool> halfEdgeProcessedNext(mesh.halfEdges.size(), false);
+  std::vector<bool> halfEdgeProcessedOpp(mesh.halfEdges.size(), false);
+  // Iterates through the halfEdges and identifies the valid ones
   for (size_t i = 0; i < mesh.halfEdges.size(); i++) {
-      if (!mesh.halfEdges[i].isDisabled() && !mesh.faces[mesh.halfEdges[i].face].isDisabled()) {
-        halfEdgeStack.push_back(i);
-      }
+    if (!mesh.halfEdges[i].isDisabled() &&
+        !mesh.faces[mesh.halfEdges[i].face].isDisabled()) {
+      halfEdgeStack.push_back(i);
     }
-    if (halfEdgeStack.size() == 0) {
-      return make_pair(manifold::Vec<manifold::Halfedge>{}, std::vector<glm::dvec3>{});
-    }
-  manifold::Vec<manifold::Halfedge> halfEdgeVec;
+  }
+  if (halfEdgeStack.size() == 0) {
+    return ConvexHull();
+  }
+  Vec<Halfedge> halfEdgeVec;
   std::vector<glm::dvec3> newVerts;
+  // To map the index of the vertex in the original point cloud to the index in
+  // the new point cloud
   std::unordered_map<size_t, size_t> vertexIndexMapping;
+  Halfedge currHalfEdge;
+  // Iterates through the valid half edges
   for (size_t index = 0; index < halfEdgeStack.size(); index++) {
-      size_t top = halfEdgeStack[index]; 
-       
-      if(mesh.halfEdges[top].isDisabled() || mesh.faces[mesh.halfEdges[top].face].isDisabled())
-      {
-        std::cout << "ERROR: Halfedge " << top << " is not disabled" << std::endl;
-        // assert(true);
-        break;
-      }
-      if (halfEdgeProcessed[top]) {
-        continue;
-      } else {
-        while(halfEdgeProcessed[top]!=true) {
-          if (mesh.halfEdges[top].opp == std::numeric_limits<size_t>::max()) {
-            std::cout << "ERROR: Halfedge " << top << " has no paired halfedge" << std::endl;
+    size_t top = halfEdgeStack[index];
+
+    if (mesh.halfEdges[top].isDisabled() ||
+        mesh.faces[mesh.halfEdges[top].face].isDisabled()) {
+      std::cout << "ERROR: Halfedge " << top << " is not disabled" << std::endl;
```

Comment:
Hmm, ideally this should fail to compile. We only include `iostream` if we're in `MANIFOLD_DEBUG` mode (it adds a lot of binary size to our WASM, etc. Let's just use our internal `ASSERT` and `DEBUG_ASSERT` functions instead.